### ISSUE TITLE
[grpc:x64-windows-static] build failure

### ISSUE THREAD

**Issue by 9ee1** [microsoft/vcpkg#23261]
**Host Environment**
- OS: Windows Server 2019 Datacenter 10.0.17763 Build 2458
- Compiler: Visual Studio Enterprise 2019 16.11.32106.194, Microsoft (R) C/C++ Optimizing Compiler Version 19.29.30139 for x64, Microsoft (R) Library Manager Version 14.29.30139.0, Microsoft (R) Incremental Linker Version 14.29.30139.0
- vcpkg Revision: https://github.com/microsoft/vcpkg/tree/6920b49047f2de6880df8ef9724cc508c5f491ca
- GitHub Actions Runner: https://github.com/actions/virtual-environments/blob/main/images/win/Windows2019-Readme.md

**To Reproduce**
1. `git checkout -b test 6920b49047f2de6880df8ef9724cc508c5f491ca`
1. `bootstrap-vcpkg.bat`
2. `vcpkg install grpc:x64-windows-static`

**Failure logs** 
Full Log File: [vcpkg-error.txt](https://github.com/microsoft/vcpkg/files/8133110/vcpkg-error.txt)

```
CMake Error at scripts/cmake/vcpkg_execute_build_process.cmake:136 (message):
Command failed: "C:/Program Files/CMake/bin/cmake.exe" --build . --config Debug --target install -- -v
Working Directory: D:/a/PROJECT/PROJECT/external/vcpkg/buildtrees/grpc/x64-windows-static-dbg
```
```
Call Stack (most recent call first):
  scripts/cmake/vcpkg_build_cmake.cmake:91 (vcpkg_execute_build_process)
  scripts/cmake/vcpkg_install_cmake.cmake:24 (vcpkg_build_cmake)
  ports/grpc/portfile.cmake:60 (vcpkg_install_cmake)
  scripts/ports.cmake:90 (include)

Package: grpc:x64-windows-static
Vcpkg version: 2020.02.04-nohash
```
```
 Creating library grpc_csharp_ext.lib and object grpc_csharp_ext.exp
grpc.lib(grpc_ares_wrapper.cc.obj) : error LNK2019: unresolved external symbol __imp_ares_library_init referenced in function "struct grpc_error * __cdecl grpc_ares_init(void)" (?grpc_ares_init@@YAPEAUgrpc_error@@XZ)
grpc.lib(grpc_ares_wrapper.cc.obj) : error LNK2019: unresolved external symbol __imp_ares_library_cleanup referenced in function "void __cdecl grpc_ares_cleanup(void)" (?grpc_ares_cleanup@@YAXXZ)
grpc.lib(grpc_ares_wrapper.cc.obj) : error LNK2019: unresolved external symbol __imp_ares_query referenced in function "void __cdecl grpc_dns_lookup_ares_continue_after_check_localhost_and_ip_literals_locked(struct grpc_ares_request *,char const *,char const *,char const *,struct grpc_pollset_set *,bool,int,class grpc_core::Combiner *)" (?grpc_dns_lookup_ares_continue_after_check_localhost_and_ip_literals_locked@@YAXPEAUgrpc_ares_request@@PEBD11PEAUgrpc_pollset_set@@_NHPEAVCombiner@grpc_core@@@Z)
grpc.lib(grpc_ares_wrapper.cc.obj) : error LNK2019: unresolved external symbol __imp_ares_search referenced in function "void __cdecl grpc_dns_lookup_ares_continue_after_check_localhost_and_ip_literals_locked(struct grpc_ares_request *,char const *,char const *,char const *,struct grpc_pollset_set *,bool,int,class grpc_core::Combiner *)" (?grpc_dns_lookup_ares_continue_after_check_localhost_and_ip_literals_locked@@YAXPEAUgrpc_ares_request@@PEBD11PEAUgrpc_pollset_set@@_NHPEAVCombiner@grpc_core@@@Z)
grpc.lib(grpc_ares_wrapper.cc.obj) : error LNK2019: unresolved external symbol __imp_ares_gethostbyname referenced in function "void __cdecl on_srv_query_done_locked(void *,int,int,unsigned char *,int)" (?on_srv_query_done_locked@@YAXPEAXHHPEAEH@Z)
grpc.lib(grpc_ares_wrapper.cc.obj) : error LNK2019: unresolved external symbol __imp_ares_parse_srv_reply referenced in function "void __cdecl on_srv_query_done_locked(void *,int,int,unsigned char *,int)" (?on_srv_query_done_locked@@YAXPEAXHHPEAEH@Z)
grpc.lib(grpc_ares_wrapper.cc.obj) : error LNK2019: unresolved external symbol __imp_ares_parse_txt_reply_ext referenced in function "void __cdecl on_txt_done_locked(void *,int,int,unsigned char *,int)" (?on_txt_done_locked@@YAXPEAXHHPEAEH@Z)
grpc.lib(grpc_ares_wrapper.cc.obj) : error LNK2019: unresolved external symbol __imp_ares_free_data referenced in function "void __cdecl on_srv_query_done_locked(void *,int,int,unsigned char *,int)" (?on_srv_query_done_locked@@YAXPEAXHHPEAEH@Z)
grpc.lib(grpc_ares_wrapper.cc.obj) : error LNK2019: unresolved external symbol __imp_ares_strerror referenced in function "struct grpc_error * __cdecl grpc_ares_init(void)" (?grpc_ares_init@@YAPEAUgrpc_error@@XZ)
grpc.lib(grpc_ares_ev_driver.cc.obj) : error LNK2001: unresolved external symbol __imp_ares_strerror
grpc.lib(grpc_ares_wrapper.cc.obj) : error LNK2019: unresolved external symbol __imp_ares_set_servers_ports referenced in function "void __cdecl grpc_dns_lookup_ares_continue_after_check_localhost_and_ip_literals_locked(struct grpc_ares_request *,char const *,char const *,char const *,struct grpc_pollset_set *,bool,int,class grpc_core::Combiner *)" (?grpc_dns_lookup_ares_continue_after_check_localhost_and_ip_literals_locked@@YAXPEAUgrpc_ares_request@@PEBD11PEAUgrpc_pollset_set@@_NHPEAVCombiner@grpc_core@@@Z)
grpc.lib(grpc_ares_wrapper.cc.obj) : error LNK2019: unresolved external symbol __imp_ares_inet_ntop referenced in function "void __cdecl on_hostbyname_done_locked(void *,int,int,struct hostent *)" (?on_hostbyname_done_locked@@YAXPEAXHHPEAUhostent@@@Z)
grpc.lib(grpc_ares_ev_driver.cc.obj) : error LNK2019: unresolved external symbol __imp_ares_init_options referenced in function "struct grpc_error * __cdecl grpc_ares_ev_driver_create_locked(struct grpc_ares_ev_driver * *,struct grpc_pollset_set *,int,class grpc_core::Combiner *,struct grpc_ares_request *)" (?grpc_ares_ev_driver_create_locked@@YAPEAUgrpc_error@@PEAPEAUgrpc_ares_ev_driver@@PEAUgrpc_pollset_set@@HPEAVCombiner@grpc_core@@PEAUgrpc_ares_request@@@Z)
grpc.lib(grpc_ares_ev_driver.cc.obj) : error LNK2019: unresolved external symbol __imp_ares_destroy referenced in function "void __cdecl grpc_ares_ev_driver_unref(struct grpc_ares_ev_driver *)" (?grpc_ares_ev_driver_unref@@YAXPEAUgrpc_ares_ev_driver@@@Z)
grpc.lib(grpc_ares_ev_driver.cc.obj) : error LNK2019: unresolved external symbol __imp_ares_cancel referenced in function "void __cdecl on_readable_locked(void *,struct grpc_error *)" (?on_readable_locked@@YAXPEAXPEAUgrpc_error@@@Z)
grpc.lib(grpc_ares_ev_driver.cc.obj) : error LNK2019: unresolved external symbol __imp_ares_getsock referenced in function "void __cdecl grpc_ares_notify_on_event_locked(struct grpc_ares_ev_driver *)" (?grpc_ares_notify_on_event_locked@@YAXPEAUgrpc_ares_ev_driver@@@Z)
grpc.lib(grpc_ares_ev_driver.cc.obj) : error LNK2019: unresolved external symbol __imp_ares_process_fd referenced in function "void __cdecl on_ares_backup_poll_alarm_locked(void *,struct grpc_error *)" (?on_ares_backup_poll_alarm_locked@@YAXPEAXPEAUgrpc_error@@@Z)
grpc.lib(grpc_ares_ev_driver_windows.cc.obj) : error LNK2019: unresolved external symbol __imp_ares_set_socket_functions referenced in function "public: virtual void __cdecl grpc_core::GrpcPolledFdFactoryWindows::ConfigureAresChannelLocked(struct ares_channeldata *)" (?ConfigureAresChannelLocked@GrpcPolledFdFactoryWindows@grpc_core@@UEAAXPEAUares_channeldata@@@Z)
grpc_csharp_ext.dll : fatal error LNK1120: 17 unresolved externals
```

**Additional context**
The build failure only occurs on the `Windows Server 2019 Datacenter 10.0.17763 Build 2458` GitHub Actions hosted runner (I linked to it above). On my development machine with `Windows Server 2019 Standard 10.0.17763 Build 2268`, `Visual Studio 2019` and `Visual Studio 2022` (latest versions at the time of this writing), the package builds fine.

Interestingly enough, this was working just fine about 6 months ago. My gut feeling is GitHub updated their `Windows Server 2019 Datacenter` image and that is now causing the build to break for some reason, but I can't figure out why.

For reasons specific to my project, I can't update to the latest version of `vcpkg` because my understanding is that would automatically bring in a newer version of `gRPC` which I can't update to at this time (if I am wrong about this, please let me know).

I am looking for guidance as to the cause of the build failure so that I can attempt to resolve it manually. I realize re-producting the error outside of a GitHub Actions runner might be difficult. I thank you in advance for your support.

**Comment by Cheney-W** [comment]
Keep active

**Comment by dg0yt** [comment]
> Keep active

Probably not, 2 years after reporting.
Probably fixed by #17230.
